src/CMakeLists.txt

```diff
@@ -291,7 +305,57 @@ if (BUILDSWIGNODE)
     ${NODE_ROOT_DIR}/include/deps/uv/include
   )
   macro(createpackagejson)
+    if (NPM)
+      # Dependency to generate wrappers before packing, skip modules that depend on 3rd party libs
+      if (NOT DEFINED reqlibname)
+        set (reqlibname "")
+      endif ()
+      if (reqlibname STREQUAL "" OR reqlibname MATCHES "mraa")
+        # Grab all module sources and headers, also add sources to gyp list
+        file (GLOB srcfiles RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}"
+                                     "${CMAKE_CURRENT_SOURCE_DIR}/*.c"
+                                     "${CMAKE_CURRENT_SOURCE_DIR}/*.cpp"
+                                     "${CMAKE_CURRENT_SOURCE_DIR}/*.cxx"
+                                     "${CMAKE_CURRENT_SOURCE_DIR}/*.h"
+                                     "${CMAKE_CURRENT_SOURCE_DIR}/*.hpp"
+                                     "${CMAKE_CURRENT_SOURCE_DIR}/*.hxx")
+        foreach (srcfile ${srcfiles})
+          file (COPY ${CMAKE_CURRENT_SOURCE_DIR}/${srcfile} DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/jsupm_${libname})
+          if (${srcfile} MATCHES ".c" OR ${srcfile} MATCHES ".cxx" OR ${srcfile} MATCHES ".cpp")
+            set (upm_LIB_SRCS_GYP "'${srcfile}',\n${upm_LIB_SRCS_GYP}")
+          endif ()
+        endforeach ()
+        set (upm_LIB_SRCS_GYP "'jsupm_${libname}JAVASCRIPT_wrap.cxx',\n${upm_LIB_SRCS_GYP}")
+
+        # Create list of include directories and copy them
+        foreach (includedir ${UPM_COMMON_HEADER_DIRS})
+          file (RELATIVE_PATH rel ${CMAKE_SOURCE_DIR} ${includedir})
+          set (upm_LIB_INCLUDE_DIRS_GYP "'${rel}',\n${upm_LIB_INCLUDE_DIRS_GYP}")
+        endforeach (includedir)
+        file (COPY ${CMAKE_SOURCE_DIR}/include DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/jsupm_${libname})
+        set (upm_LIB_INCLUDE_DIRS_GYP "'.',\n${upm_LIB_INCLUDE_DIRS_GYP}")
+
+        # Utilities and interfaces
+        file (COPY ${CMAKE_SOURCE_DIR}/src/utilities DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/jsupm_${libname})
+        file (COPY ${CMAKE_SOURCE_DIR}/src/upm DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/jsupm_${libname})
+        set (upm_LIB_SRCS_GYP "'utilities/upm_utilities.c',\n${upm_LIB_SRCS_GYP}")
+        set (upm_LIB_INCLUDE_DIRS_GYP "'utilities',\n${upm_LIB_INCLUDE_DIRS_GYP}")
+        set (upm_LIB_INCLUDE_DIRS_GYP "'upm',\n${upm_LIB_INCLUDE_DIRS_GYP}")
+
+        # Add readme, package.json for NPM and node-gyp config file
+        configure_file (${PROJECT_SOURCE_DIR}/src/binding.gyp.in ${CMAKE_CURRENT_BINARY_DIR}/jsupm_${libname}/binding.gyp @ONLY)
+        configure_file (${PROJECT_SOURCE_DIR}/src/package.json.npm.in ${CMAKE_CURRENT_BINARY_DIR}/jsupm_${libname}/package.json @ONLY)
+        configure_file (${PROJECT_SOURCE_DIR}/doxy/README.npm.md ${CMAKE_CURRENT_BINARY_DIR}/jsupm_${libname}/README.md)
+
+        add_dependencies (npmpkg jsupm_${libname})
+      else ()
+          message ("Skipping jsupm_${libname}, needs ${reqlibname}")
```

Comment:
Makes it easier when using a visual interface to cmake (e.g. ccmake), depends on preference. With STATUS they only briefly show on the bottom "status" bar and don't show up in the end report of the configuration step.